R/assign.R

```diff
@@ -0,0 +1,243 @@
+#'
+#' @returns The returned data set depends on the value of `tgt_dat`:
+#' - If no target dataset is supplied, meaning that `tgt_dat` defaults to
+#' `NULL`, then the returned data set is `raw_dat`, selected for the variables
+#' indicated in `id_vars`, and a new extra column: the derived variable, as
+#' indicated in `tgt_var`.
+#' - If the target dataset is provided, then it is merged with the raw data set
+#' `raw_dat` by the variables indicated in `id_vars`, with a new column: the
+#' derived variable, as indicated in `tgt_var`.
+#'
+#' @examples
+#'
+#' md1 <-
+#'   tibble::tibble(
+#'     oak_id = 1:14,
+#'     raw_source = "MD1",
+#'     patient_number = 101:114,
+#'     MDIND = c(
+#'       "NAUSEA", "NAUSEA", "ANEMIA", "NAUSEA", "PYREXIA",
+#'       "VOMITINGS", "DIARHHEA", "COLD",
+#'       "FEVER", "LEG PAIN", "FEVER", "COLD", "COLD", "PAIN"
+#'     )
+#'   )
+#'
+#' assign_no_ct(
+#'   raw_dat = md1,
+#'   raw_var = "MDIND",
+#'   tgt_var = "CMINDC",
+#' )
+#'
+#' cm_inter <-
+#'   tibble::tibble(
+#'     oak_id = 1:14,
+#'     raw_source = "MD1",
+#'     patient_number = 101:114,
+#'     CMTRT = c(
+#'       "BABY ASPIRIN",
+#'       "CORTISPORIN",
+#'       "ASPIRIN",
+#'       "DIPHENHYDRAMINE HCL",
+#'       "PARCETEMOL",
+#'       "VOMIKIND",
+#'       "ZENFLOX OZ",
+#'       "AMITRYPTYLINE",
+#'       "BENADRYL",
+#'       "DIPHENHYDRAMINE HYDROCHLORIDE",
+#'       "TETRACYCLINE",
+#'       "BENADRYL",
+#'       "SOMINEX",
+#'       "ZQUILL"
+#'     ),
+#'     CMROUTE = c(
+#'       "ORAL",
+#'       "ORAL",
+#'       NA,
+#'       "ORAL",
+#'       "ORAL",
+#'       "ORAL",
+#'       "INTRAMUSCULAR",
+#'       "INTRA-ARTERIAL",
+#'       NA,
+#'       "NON-STANDARD",
+#'       "RANDOM_VALUE",
+#'       "INTRA-ARTICULAR",
+#'       "TRANSDERMAL",
+#'       "OPHTHALMIC"
+#'     )
+#'   )
+#'
+#' # Controlled terminology specification
+#' (ct <- read_ct_example("ct-01-cm"))
+#'
+#' assign_ct(
+#'   raw_dat = md1,
+#'   raw_var = "MDIND",
+#'   tgt_var = "CMINDC",
+#'   ct = ct,
+#'   cl = "C66729",
+#'   tgt_dat = cm_inter
+#' )
+#'
+#' @name assign
+NULL
+
+#' @order 1
+#' @export
+#' @rdname assign
+assign_no_ct <- function(raw_dat,
+                         raw_var,
+                         tgt_var,
+                         tgt_dat = NULL,
+                         id_vars = oak_id_vars()) {
+  admiraldev::assert_character_scalar(raw_var)
+  admiraldev::assert_character_scalar(tgt_var)
+  admiraldev::assert_character_vector(id_vars)
+  assertthat::assert_that(contains_oak_id_vars(id_vars),
+    msg = "`id_vars` must include the oak id vars."
+  )
+  admiraldev::assert_data_frame(raw_dat, required_vars = rlang::syms(c(id_vars, raw_var)))
+  admiraldev::assert_data_frame(tgt_dat, required_vars = rlang::syms(id_vars), optional = TRUE)
+
+  sdtm_assign(
+    raw_dat = raw_dat,
+    raw_var = raw_var,
+    tgt_var = tgt_var,
+    tgt_dat = tgt_dat,
+    id_vars = id_vars
+  )
+}
+
+#' @order 2
+#' @export
+#' @rdname assign
```

Comment:
Should we add examples for `assign_ct` as well?

Reply:
There are already. Take a look at line 177.